### PR TITLE
[Site] restructure site components to enable building guidelines pages

### LINKED ISSUE #225: [Site] create Guidelines pages template
As a developer I would like to start the conversation between stakeholders about what guidelines/principles pages should be surfaced . We need to have a separate layout for Guide pages.

AC:
- when someone visits `/guidelines/:somepage` a different layout is rendered from the React Router
- template is not restricted to just prose content, but allows for variations like color swatches, design tokens, etc.

### PR DESCRIPTION
**Pull request by riley**
### Description
This PR is mostly structural changes and filler content to set up later development of Guidelines pages.

closes #225 

- created a `/packages/site/src/pages` folder that will hold some (all?) of the top-level content
- stopgap `SideNav.js` component

We can merge this into `master` and even publish the site if we so choose; the router has not been changed and the site still goes from `/` to `/components/button`.

### Motivation and context
We need to block out areas were we need content provided to us by other teams. In lieu of design comps, we can at least create the IA for the demo site and copy can be refined as we go along. Creating pages will get the conversation moving about what content needs to go where.

### Screenshots or videos, if appropriate
The extra pages are currently quite ugly, so I'm going to hide behind this door 🚪 

### How to test
1. fire up the site like normal
2. navigate to `/guidelines`
3. You should see a Table of Contents page. (do we need one? that's what we're here to figure out!). Other pages:
 - Color
 - Theming / Tokens
 - Typography
 - Getting Started
4. check out the `packages/site/src` folder structure. Is this what we want to have? I'm open to other better ideas.

### Types of changes
applicable. Note: Whatever you choose here should match your commit messages. -->
- New Feature

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated

### How does this PR make you feel?

there is no cow level

### CODE REVIEW

File: packages/site/src/components/App.js [hunk 1]
```diff
@@ -72,12 +74,22 @@ export default function App({ className, demos }: Props) {
 
   return (
     <Root className={className}>
-      <ThemeProvider theme={{ backgroundColor: siteTheme.color_grayLight }}>
-        <Nav demos={siteDemos} />
-      </ThemeProvider>
+      {/* this could be rendered inside of ComponentDoc. what's the best way? */}
```
Comment (kylegach) [inline packages/site/src/components/App.js]: This feels better, imo.

File: packages/site/src/components/App.js [hunk 2]
```diff
@@ -17,17 +17,19 @@
...
-} from '@mineral-ui/component-utils';
+import { createStyledComponent } from '@mineral-ui/component-utils';
 import createKeyMap from '../utils/createKeyMap';
 import ComponentDoc from './ComponentDoc';
 import Footer from './Footer';
 import _Nav from './Nav';
-import siteTheme from './siteTheme';
 import styleReset from './styleReset';
 
+import GettingStarted from '../pages/GettingStarted';
```
Comment (brentertz) [inline packages/site/src/components/App.js]: If you are going to always be importing all of the pages, you could add a pages/index, and just import that here.

File: packages/site/src/components/App.js [hunk 3]
```diff
@@ -17,17 +17,19 @@
...
+import Color from '../pages/Color';
+import TableOfContents from '../pages/TableOfContents';
+import Tokens from '../pages/Tokens.js';
```
Comment (brentertz) [inline packages/site/src/components/App.js]: Unneeded `.js` extension

File: packages/site/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  letterSpacing: '-0.2em',
+
+  '& > :first-child': {
+    display: 'block',
+    marginRight: 20,
+    borderRadius: 40,
+    width: 80,
+    height: 80,
+    backgroundImage:
+      'url(http://www.treasuremountainmining.com/image/Wulfenite.jpg)',
```
Comment (brentertz) [inline packages/site/src/components/Header.js]: Please do not link to an image on someone else's site, unless it is a site designed for that, or they give you permission.

File: packages/site/src/utils/guides.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+export default {
```
Comment (brentertz) [inline packages/site/src/utils/guides.js]: Is this file expected to live here long term or is it for mock-up purposes only?  If it is long term, than it is probably not a "util".  Also, consider having a separate title or name field from the keys and avoid having spaces in the keys.

File: packages/site/src/components/GuideContent.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
```
Comment (brentertz) [inline packages/site/src/components/GuideContent.js]: You need to run `npm run format` as none of these new files have the license header.

**Comment by riley** [comment]
@brentertz We don't really need it to be in `master` since we'll have deploy previews thru Netlify once #249 is merged. The important part is some way to share this with stakeholders and the flexibility to iterate quickly. I agree that merging to `master` requires a certain degree of rigor that the ideation process cannot sustain.

@kylegach @coldpour what are your thoughts about having a long-living `website` branch that gets shared around privately via deploy previews until we have an MVP?

**Comment by riley** [comment]
closing in favor of #256

**Pull request closed without merging**